slackify-html.js

```diff
@@ -1,27 +1,46 @@
-var htmlparser = require("htmlparser"),
-  Entities = require("html-entities").AllHtmlEntities;
-
-entities = new Entities();
+const htmlparser = require("htmlparser"),
+    entities = require("html-entities");
 
 module.exports = function slackify(html) {
-  var handler = new htmlparser.DefaultHandler(function (error, dom) {
+  const handler = new htmlparser.DefaultHandler(function (error, dom) {
     // error ignored
   });
-  var parser = new htmlparser.Parser(handler);
+  const parser = new htmlparser.Parser(handler);
   parser.parseComplete(html);
-  var dom = handler.dom;
+  const dom = handler.dom;
   if (dom) return entities.decode(walk(dom));
   else return "";
 };
 
+function walkLink(dom) {
+  let out='';
+
+  if (dom) {
+    dom.forEach(function (el) {
+      if (el.type === 'text')
+        out += el.data;
+      else if (el.type === 'tag' && el.children)
+        out += walkLink(el.children);
+    });
+  }
+  return out;
+}
+
```

Comment:
This is the main change, a function that skips over tags found in a link text value until it finds the text.